### ISSUE TITLE
daemon skitz moment - "Final must be initialized" false positive occasionally appears and disappears when unrelated changes are made

### ISSUE THREAD

**Issue by DetachHead** [KotlinIsland/basedmypy#863]
```py
class Foo(ABC):
    @property
    @abstractmethod
    def relation(self) -> Bar: ...

    objects: Final[Collection[Baz]]

    def __init__(self, objects: Collection[Baz] | Baz):
        self.objects = [objects] if isinstance(objects, str | int) else objects
```

2.9.1
